obspy/io/ascii/core.py

```diff
@@ -325,17 +327,15 @@ def _write_slist(stream, filename, custom_fmt=None,  **kwargs):  # @UnusedVariab
             if trace.data.dtype.name.startswith('int'):
                 dtype = 'INTEGER'
                 fmt = '%d'
-            elif trace.data.dtype.name.startswith('float32'):
+            elif trace.data.dtype.name.startswith('float'):
                 dtype = 'FLOAT'
-                fmt = '%+.8g'
-            elif trace.data.dtype.name.startswith('float64'):
-                dtype = 'FLOAT'
-                fmt = '%+.10g'
+                fmt = '%+.10e'
 
             else:
                 raise NotImplementedError
             # fmt
             if  custom_fmt is not None:
+                dtype = 'CUSTOM'
```

Comment:
I really don't care about this `CUSTOM` case a lot (I still don't see the point writing a file that can not be read afterwards), but I think if the custom format is something that relates to float (`%....f`, `%...g`, ...) than the type should also be `FLOAT`, so that it can be read again (same goes for custom integer types obviously).

Just my two cents. Leave as is if you like.

Reply:
@megies: this argument is invalid - you can provide any string format - such as ```%s```, ```%04d``` or ```%x``` - so ```FLOAT``` is not useful at all

Reply:
What about creating a little method that's output will be FLOAT for `%..e f g` and INTEGER for `%..d i u`?